homeassistant/components/matrix/__init__.py

```diff
@@ -52,17 +56,20 @@
     cv.has_at_least_one_key(CONF_WORD, CONF_EXPRESSION),
 )
 
-CONFIG_SCHEMA = vol.Schema(
+ROOM_SCHEMA: vol.Schema = vol.All(
+    cv.ensure_list,
+    [cv.string],
+)
+
+CONFIG_SCHEMA: vol.Schema = vol.Schema(
```

Comment:
This schema needs to be marked deprecated.

Reply:
Will do. But if it's marked deprecated, I will have to add `rooms` and `commands` settings to the options flow.

Reply:
```yaml
# Example configuration.yaml entry
matrix:
  homeserver: https://matrix.org
  username: "@my_matrix_user:matrix.org"
  password: supersecurepassword
  rooms:
    - "#hasstest:matrix.org"
  commands:
    - word: my_command
      name: my_command
```

How can I handle `rooms` and `commands` in the options flow? Is it possible to let the user type JSON strings? I see `ble_monitor` uses a more complex method to add/delete bluetooth devices, I am wondering if I will have to do so.